hooks/mouseEvent.ts

```diff
@@ -0,0 +1,40 @@
+import { useCallback, useRef, useState } from 'react'
+import { offsetLeft } from 'utils/dom'
+
+export const useMouseMoveListener = <T extends HTMLElement>() => {
+  const [diff, setDiff] = useState(0)
+  const mouseRef = useRef<T>(null)
+
+  const onMouseMove = useCallback((e: TouchEvent | MouseEvent) => {
+    e.stopPropagation()
+
+    if (e instanceof TouchEvent) {
+      const { clientX } = e.changedTouches[0]
+      const diff = clientX - offsetLeft(mouseRef.current)
+      setDiff(diff)
```

Comment:
how about early return?
```suggestion
      setDiff(diff)
      
      return
```